### DISCUSSION TITLE
reducing-the-size-of-a-git-repository-with-git-replace/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#373] [Announcements]
# reducing-the-size-of-a-git-repository-with-git-replace/

In this post I show how you can split a git repo into 'current' and 'history' repos, and then join them again using git-replace as necessary

https://andrewlock.net/reducing-the-size-of-a-git-repository-with-git-replace/

[a comment by an automated account is not shown]

**Comment by MateuszStefek** [comment]
--rebase-merges  will work in such a way, that git will ask you to resolve the same merge conflicts you already solved in the original merges. Luckily, we already know the exact file content we expect in the merge commit. We can reset soft the files to the desired commit. The only problem is that git reset isn't available during rebase.

If a rebase fails with a message like, this:
```
Could not apply 6aad7d5ff62... # Merge commit message
```

you can resolve the problem with this trick:
```
TARGET_COMMIT=6aad7d5ff62

# Just commit without resolving any conflicts
git add .
git commit -m "."

# Apply the desired file structure, forcefully
git diff -u HEAD $TARGET_COMMIT | patch -p1

# Amend the current merge commit (detached head)
git add .
git commit --amend --no-edit

# Rinse and repeat until you reach the end of your rebase
 git rebase --continue

```

**Comment by omatrot** [comment]
Excellent post :) To push to a new git repo in the first part, an example would have been a plus.
Alors your dealing with only the main branch whereas I have current development branch on my 1GB+ repo. Is there a risk that they would be lost during the rewrite? Thanks for your clarification.
